tendrl/common/atoms/base_atom.py

```diff
@@ -0,0 +1,11 @@
+import abc
+
+
+class BaseAtom(object):
+    __metaclass__ = abc.ABCMeta
+
+    @abc.abstractmethod
+    def run(self, parameters):
+        raise NotImplementedError(
+            'define the function run to use this class'
```

Comment:
Inherit "NotImplementedError" and create "AtomNotImplementedError" (common/atoms/exceptions.py) and provide msg "run function not implemented"